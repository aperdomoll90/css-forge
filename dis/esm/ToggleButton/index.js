export { ToggleButton } from "./ToggleButton";
export * from './ToggleButton.types';
//# sourceMappingURL=index.js.map