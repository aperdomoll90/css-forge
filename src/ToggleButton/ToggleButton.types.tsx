import { MenuItemsArrayPropsTypes } from '../utils/GlobalTypes.types'

export interface ToggleButtonPropsType {
  MenuItemsArray: MenuItemsArrayPropsTypes[]
  color?: `#${string}` | `rgb(${number}, ${number}, ${number})` | `rgba(${number}, ${number}, ${number}, ${number})` | ''
  buttonHover?: string
  labelColor?: string
  labelColorHover?: string
  buttonBackgroundColor?: string
  menuBackgroundColor?: string
}
