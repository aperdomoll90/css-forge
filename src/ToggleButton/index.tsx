export { ToggleButton } from "./ToggleButton"
export * from './ToggleButton.types'