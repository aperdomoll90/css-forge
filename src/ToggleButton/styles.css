
.ToggleButton {
  width: var(--size);
  aspect-ratio: 1;
  position: absolute;
  top: var(--top);
  bottom: var(--bottom);
  left: var(--left);
  right: var(--right);
  display: flex;
  justify-content: center;
  align-items: center;
  background: var(--buttonBackgroundColor);
  border-radius: 5px;
  box-shadow: 0 10px 20px var(--shadow);
  overflow: hidden;
  cursor: pointer;
  transition: 0.5s ease-in-out;
  z-index: 9999;
  border: none;
  outline: none;
}
.ToggleButton:hover {
  background: var(--buttonHover);
}

.ToggleButton span {
  position: absolute;
  width: 80%;
  height: 4px;
  border-radius: 4px;
  background: var(--color);
  transition: 0.5s ease-in-out;
}

.button-sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border-width: 0;
}

.ToggleButton span:nth-child(1) {
  width: 50%;
  left: 12%;
  transform: translateY(calc(var(--size) / -3.5));
}
.ToggleButtonActive span:nth-child(1) {
  width: 75%;
  transform: translateY(0px) rotate(45deg);
  transition-delay: 0.125s;
}

.ToggleButton span:nth-child(2) {
  width: 35%;
  left: 12%;
  transform: translateY(calc(var(--size) / 3.5));
}
.ToggleButtonActive span:nth-child(2) {
  width: 75%;
  transform: translateY(0px) rotate(315deg);
  transition-delay: 0.25s;
}

.ToggleButtonActive span:nth-child(3) {
  transform: translateX(120%);
}
