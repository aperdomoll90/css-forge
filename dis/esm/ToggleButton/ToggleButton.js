var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect } from 'react';
import './styles.css';
import '../utils/GlobalStyles.css';
import { LightenDarkenColor } from '../utils/ColorManipulation';
export var ToggleButton = function (_a) {
    var size = _a.size, color = _a.color, buttonHover = _a.buttonHover, buttonBackgroundColor = _a.buttonBackgroundColor, shadow = _a.shadow, setActive = _a.setActive, ariaControls = _a.ariaControls, ariaExpanded = _a.ariaExpanded, top = _a.top, bottom = _a.bottom, left = _a.left, right = _a.right, customClass = _a.customClass;
    var defaultColor = '#fff';
    var defaultBackgroundColor = '#303030da';
    var stylesProps = {
        '--size': size ? "".concat(size, "rem") : '2rem',
        '--color': color ? color : defaultColor,
        '--buttonHover': buttonHover ? buttonHover : LightenDarkenColor(buttonBackgroundColor || defaultBackgroundColor, -80),
        '--buttonBackgroundColor': buttonBackgroundColor ? buttonBackgroundColor : defaultBackgroundColor,
        '--shadow': shadow ? 'rgba(0, 0, 0, 0.8)' : '',
        '--top': top ? "".concat(top, "rem") : '1rem',
        '--bottom': bottom ? "".concat(bottom, "rem") : 'auto',
        '--left': left ? "".concat(left, "rem") : 'auto',
        '--right': right ? "".concat(right, "rem") : '1rem',
    };
    var toggleMenu = function () { return setActive(function (value) { return !value; }); };
    useEffect(function () {
        var menuToggle = document.querySelector('.ToggleButton');
        if (menuToggle) {
            menuToggle.onclick = function () {
                menuToggle.classList.toggle('ToggleButtonActive');
                toggleMenu();
            };
        }
    }, []);
    var classArray = customClass && customClass ? "ToggleButton ".concat(customClass) : 'ToggleButton';
    return (_jsxs("button", __assign({ className: classArray, "aria-controls": ariaControls, "aria-expanded": ariaExpanded, style: stylesProps }, { children: [_jsx("span", {}), _jsx("span", {}), _jsx("span", {}), _jsx("span", __assign({ className: 'sr-only' }, { children: "menu" }))] })));
};
//# sourceMappingURL=ToggleButton.js.map