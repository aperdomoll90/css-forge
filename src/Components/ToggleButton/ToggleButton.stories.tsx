import { Meta, StoryFn } from '@storybook/react'
import ToggleButton from './ToggleButton'
import StorybookContainer from '../StorybookContainer'

export default {
  title: 'Buttons',
  component: ToggleButton,
  parameters: {
    jest: ['ToggleButton.test.tsx'],
  },
} as Meta

const ToggleButtonTemplate: StoryFn = () => (
  <StorybookContainer>
    <ToggleButton />
  </StorybookContainer>
)

export const ToggleButtonComponent = ToggleButtonTemplate.bind({})
