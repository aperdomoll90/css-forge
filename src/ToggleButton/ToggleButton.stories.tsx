import { Meta, StoryFn } from '@storybook/react'
import { ToggleButton } from './ToggleButton'
import StorybookContainer from '../StorybookContainer'
import { ToggleButtonPropsType } from './ToggleButton.types'
import { MenuItemsArrayPropsTypes } from '../utils/GlobalTypes.types'
import { BarIcon, CameraIcon, CodeIcon, GearIcon, HomeIcon, MapIcon, PawIcon, PlanetIcon, ProfileIcon } from '../IconCollection/IconCollection'

export default {
  title: 'Buttons',
  component: ToggleButton,
  parameters: {
    jest: ['ToggleButton.test.tsx'],
  },
  argTypes: {
    color: { options: ['', 'black', 'white', 'red', 'blue'], control: { type: 'radio' } },
    buttonHover: { options: ['', 'black', 'white', 'red', 'blue'], control: { type: 'radio' } },
    labelColor: { options: ['', 'black', 'white', 'red', 'blue'], control: { type: 'radio' } },
    labelColorHover: { options: ['', 'black', 'white', 'red', 'blue'], control: { type: 'radio' } },
    buttonBackgroundColor: { options: ['', 'black', 'white', 'red', 'blue'], control: { type: 'radio' } },
    menuBackgroundColor: { options: ['', 'black', 'white', 'red', 'blue'], control: { type: 'radio' } },
  },
} as Meta

const MenuItemsArray: MenuItemsArrayPropsTypes[] = [
  {
    label: 'Home',
    icon: HomeIcon,
    link: '#',
  },
  {
    label: 'Settings',
    icon: GearIcon,
    link: '#',
  },
  {
    label: 'Analytics',
    icon: BarIcon,
    link: '#',
  },
  {
    label: 'Profile',
    icon: ProfileIcon,
    link: '#',
  },
  {
    label: 'Code',
    icon: CodeIcon,
    link: '#',
  },
]

const ToggleButtonTemplate: StoryFn<ToggleButtonPropsType> = args => (
  <StorybookContainer>
    <ToggleButton {...args} />
  </StorybookContainer>
)

export const ToggleButtonComponent = ToggleButtonTemplate.bind({})
ToggleButtonComponent.args = {
  MenuItemsArray: MenuItemsArray,
}
