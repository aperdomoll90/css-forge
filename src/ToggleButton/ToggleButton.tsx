import "./styles.css";
import "../utils/GlobalStyles.css";
import { ToggleButtonPropsType } from "./ToggleButton.types";
import { LightenDarkenColor } from "../utils/ColorManipulation";

export const ToggleButton: React.FC<ToggleButtonPropsType> = ({
  size,
  color,
  buttonHover,
  buttonBackgroundColor,
  shadow,
  active,
  setActive,
  ariaControls,
  ariaExpanded,
  top,
  bottom,
  left,
  right,
  customClass,
}) => {
  const defaultColor = "#fff";
  const defaultBackgroundColor = "#303030da";

  const stylesProps = {
    "--size": size ? `${size}rem` : "2rem",
    "--color": color ? color : defaultColor,
    "--buttonHover": buttonHover
      ? buttonHover
      : LightenDarkenColor(
          buttonBackgroundColor || defaultBackgroundColor,
          -80
        ),
    "--buttonBackgroundColor": buttonBackgroundColor
      ? buttonBackgroundColor
      : defaultBackgroundColor,
    "--shadow": shadow ? "rgba(0, 0, 0, 0.8)" : "",
    "--top": top ? `${top}rem` : "auto",
    "--bottom": bottom ? `${bottom}rem` : "auto",
    "--left": left ? `${left}rem` : "auto",
    "--right": right ? `${right}rem` : "auto",
  };

  const isActive = active ? "ToggleButtonActive ToggleButton" : "ToggleButton";
  const classArray =
    customClass && customClass ? `${isActive} ${customClass}` : isActive;

  return (
    <button
      className={classArray}
      aria-controls={ariaControls}
      aria-expanded={ariaExpanded}
      style={stylesProps as React.CSSProperties}
      onClick={() => setActive()}
    >
      <span></span>
      <span></span>
      <span></span>
      <span className="sr-only">menu</span>
    </button>
  );
};
