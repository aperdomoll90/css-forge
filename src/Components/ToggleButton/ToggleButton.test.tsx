import { render } from '@testing-library/react'
import ToggleButton from './ToggleButton'

describe('ToggleButton component', () => {
  // const { axe, toHaveNoViolations } = require('jest-axe')
  // expect.extend(toHaveNoViolations)

  const renderComponent = () => render(<ToggleButton />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
  //   it('should demonstrate ADA compliance on this component', async () => {
  //     const { container } = renderComponent()
  //     const results = await axe(container)
  //     expect(container).toHaveNoViolations()
  //   })
})
