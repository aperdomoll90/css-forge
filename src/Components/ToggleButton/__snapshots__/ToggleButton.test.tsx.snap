// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton component should match snapshots 1`] = `
<div>
  <div
    class="fieldLocation"
  >
    <div
      class="ToggleButtonWrapper"
    >
      <div
        class="ToggleButton"
      >
        <span />
        <span />
        <span />
      </div>
      <div
        class="ToggleButtonDropDown"
      >
        <p>
          <ion-icon
            name="home-outline"
          />
           Home
        </p>
        <p>
          <ion-icon
            name="chatbubble-outline"
          />
           Messages
        </p>
        <p>
          <ion-icon
            name="person-outline"
          />
           Profile
        </p>
        <p>
          <ion-icon
            name="settings-outline"
          />
           Settings
        </p>
      </div>
    </div>
  </div>
</div>
`;
